src/legacy/amos/zabs.f

```diff
@@ -16,7 +16,7 @@ DOUBLE PRECISION FUNCTION AZABS(ZR, ZI)
 C     TRUE FLOATING ZERO
 C-----------------------------------------------------------------------
       S = S*1.0D+0
-      IF (S.EQ.0.0D+0) GO TO 20
+      IF ((S-0.0D+0).LE.0.0D+0) GO TO 20
```

Comment:
This change seems wrong --- if `S` is zero, which happens if `ZR` and `ZI` are zero, then the original condition will be true, but the new condition will never be true, because `S` is always positive or zero. I would use the `abs` idiom here.

Reply:
Thank you. I added [commit](https://github.com/certik/fortran-utils/pull/24/commits/d99482ae027f83791a6bc67cc590b7a2e1978a7b). I used DABS, because it is for double precision.